models/ht_neuron.cpp

```diff
@@ -191,17 +169,76 @@ ht_neuron_dynamics( double, const double y[], double f[], void* pnode )
   const double tau_h_T = 8.2
     + ( 56.6 + 0.27 * std::exp( ( V + 115.2 ) / 5.0 ) )
       / ( 1.0 + std::exp( ( V + 86.0 ) / 3.2 ) );
-  f[ S::IT_m ] = ( m_inf_T - y[ S::IT_m ] ) / tau_m_T;
-  f[ S::IT_h ] = ( h_inf_T - y[ S::IT_h ] ) / tau_h_T;
+  f[ S::m_IT ] = ( node.m_eq_T_( V ) - y[ S::m_IT ] ) / tau_m_T;
+  f[ S::h_IT ] = ( node.h_eq_T_( V ) - y[ S::h_IT ] ) / tau_h_T;
 
   // I_h
   const double tau_m_h =
     1.0 / ( std::exp( -14.59 - 0.086 * V ) + std::exp( -1.87 + 0.0701 * V ) );
-  f[ S::Ih_m ] = ( m_inf_h - y[ S::Ih_m ] ) / tau_m_h;
+  f[ S::m_Ih ] = ( node.m_eq_h_( V ) - y[ S::m_Ih ] ) / tau_m_h;
 
   return GSL_SUCCESS;
 }
 
+inline double
+nest::ht_neuron::m_eq_h_( double V ) const
+{
+  const double I_h_Vthreshold = -75.0;
+  return 1.0 / ( 1.0 + std::exp( ( V - I_h_Vthreshold ) / 5.5 ) );
+}
+
+inline double
+nest::ht_neuron::h_eq_T_( double V ) const
+{
+  return 1.0 / ( 1.0 + std::exp( ( V + 83.0 ) / 4 ) );
+}
+
+inline double
+nest::ht_neuron::m_eq_T_( double V ) const
+{
+  return 1.0 / ( 1.0 + std::exp( -( V + 59.0 ) / 6.2 ) );
+}
+
+inline double
+nest::ht_neuron::D_eq_KNa_( double V ) const
+{
+  const double D_influx_peak = 0.025;
+  const double D_thresh = -10.0;
+  const double D_slope = 5.0;
+  const double D_eq_0 = 0.001;
+
+  const double D_influx =
+    D_influx_peak / ( 1.0 + std::exp( -( V - D_thresh ) / D_slope ) );
+  return P_.tau_D_KNa * D_influx + D_eq_0;
```

Comment:
I hadn't thought about this, but you are right. The dimension of D does not matter, as long as it is the same as of d_1/2, since the two are divided in the end, so dimensions drop out. I suggest to leave D and d_1/2 dimensionless, since they represent some "factor" which has no immediate physical counterpart (but models some sort of ion concentration). Then D_eq also is dimensionless, while D_influx and D_influx,peak must have units of inverse time, explicitly 1/ms. This makes sense, since these values essentially represent ion concentration *change*. I have updated the notebook.